src/operation.rs

```diff
@@ -473,6 +492,68 @@ impl Operation {
                     .send(Message::PendingProgress(id, 100.0))
                     .await;
             }
+            Self::Extract { paths, to } => {
```

Comment:
ZipArchive also has the method extract,which also handles symlinks.

Reply:
`ZipArchive::`[`extract()`](https://docs.rs/zip/1.1.4/zip/read/struct.ZipArchive.html#method.extract) what disappeared from my review. Especially as current code in this PR does directory extraction twice.